plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraMetadata.java

```diff
@@ -209,7 +230,7 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorSession session, ConnectorTableHandle tableHandle, Constraint constraint)
     {
         CassandraTableHandle handle = (CassandraTableHandle) tableHandle;
-        if (handle.getPartitions().isPresent() || !handle.getClusteringKeyPredicates().isEmpty()) {
+        if (handle.isSynthetic() || handle.getPartitions().isPresent() || !handle.getClusteringKeyPredicates().isEmpty()) {
```

Comment:
Can we have a separate `if` statement for this ? like JdbcBased connector we can also add the filter as as additional filter criteria right ? And we could have a dedicated //TODO for the same.